src/ui/tests/orders/criticalPath/replaseAddDeleteProduct.spec.ts

```diff
@@ -0,0 +1,118 @@
+import { TAGS } from 'data/testTags.data';
+import { expect, test } from 'fixtures/ordersCustom.fixture';
+
+test.describe('[UI] [Orders] [Orders Details] [Edit Products] Replace/add/delete product', () => {
+  let productNames: string[] = [];
+  let orderId: string;
+
+  test.beforeEach(
+    async ({
+      homeUIService,
+      orderDraftStatus,
+      ordersPage,
+      orderDetailsPage,
+      productsApiService,
+      signInApiService,
+    }) => {
+      const PRODUCTS_TO_CREATE_COUNT = 3;
+      const result = await orderDraftStatus(PRODUCTS_TO_CREATE_COUNT);
+      orderId = result.id;
+
+      const token = await signInApiService.loginAsLocalUser();
+
+      const products = await Promise.all(
+        result.productsIds.map((productId) =>
+          productsApiService.getById(token, productId),
+        ),
+      );
+
+      productNames = products.map((p) => p.name);
+
+      await homeUIService.openAsLoggedInUser();
+      await homeUIService.openModule('Orders');
+
+      await ordersPage.clickDetailsButton(orderId);
+      await orderDetailsPage.waitForOpened();
+
+      await orderDetailsPage.receivedProductsSection.clickEditProductsPencilButton();
+      await orderDetailsPage.waitForOpened();
+    },
+  );
+
+  test(
+    'Replace the first product with the second',
+    { tag: [TAGS.ORDERS] },
+    async ({ orderDetailsPage }) => {
+      const firstProductName = productNames[0];
+      const secondProductName = productNames[1];
+
+      await orderDetailsPage.editOrderModal.selectProductAtPosition(
+        secondProductName,
+        1,
+      );
+
+      expect(await orderDetailsPage.editOrderModal.isSaveEnabled()).toBe(true);
+
+      await orderDetailsPage.editOrderModal.clickSave();
+      await orderDetailsPage.waitForSpinner();
+
+      const updatedProductNames =
+        await orderDetailsPage.receivedProductsSection.getAllProductNames();
+      expect(updatedProductNames).toContain(secondProductName);
+      expect(updatedProductNames).not.toContain(firstProductName);
+    },
+  );
+
+  test(
+    'Add the same product as the second one',
+    { tag: [TAGS.ORDERS] },
+    async ({ orderDetailsPage }) => {
+      const firstProductName = productNames[0];
+
+      await orderDetailsPage.editOrderModal.clickAddProduct();
+
+      const productCount =
+        await orderDetailsPage.editOrderModal.productsList.count();
+
+      await orderDetailsPage.editOrderModal.selectProductAtPosition(
+        firstProductName,
+        productCount,
+      );
+
+      expect(await orderDetailsPage.editOrderModal.isSaveEnabled()).toBe(true);
+
+      await orderDetailsPage.editOrderModal.clickSave();
+      await orderDetailsPage.waitForSpinner();
+
+      const updatedProductNames =
+        await orderDetailsPage.receivedProductsSection.getAllProductNames();
+
+      expect(
+        updatedProductNames.filter((name) => name === firstProductName).length,
+      ).toBeGreaterThan(1);
```

Comment:
Мы же знаем, что продуктов должно быть два с таким именем
```suggestion
      ).toEqual(2);
```